### PR TITLE
outdated: Sorted the package list alphabetically

### PR DESCRIPTION
**Pull request by watilde**
Fixes: https://github.com/npm/npm/issues/8401

### CODE REVIEW

**Comment by othiym23** [comment]
Rebased and landed by @zkat as 31aada4ccc369c0903ff7f233f464955d12c6fe2. @zkat also wrote some tests for this change as f42ba69c82890a9ed872f2800d865f8f205496f9. Thanks once more / as always watildeさん!

image

**Comment by watilde** [comment]
@zkat Thank you so much for some tests you wrote!

@othiym23 Thanks for your kind comment. What a amazing kawaii cats picture :heart_eyes_cat:

File: lib/outdated.js [hunk 1]
```diff
@@ -48,6 +48,12 @@ function outdated (args, silent, cb) {
   outdated_(args, dir, {}, 0, function (er, list) {
     if (!list) list = []
     if (er || silent || list.length === 0) return cb(er, list)
+    list.sort(function(a, b) {
+      var aa = a[1].toLowerCase()
+        , bb = b[1].toLowerCase()
+      return aa === bb ? 0
```
Comment (thefourtheye) [inline lib/outdated.js]: Why only first characters are compared? This will not sort properly.

Comment (othiym23) [inline lib/outdated.js]: Good catch – if you wanted to patch this to be

```javascript
var aa = a.toLowerCase()
var bb = b.toLowerCase()
return aa == bb ? 0 : aa < bb ? -1 : 1
```
as well as a test that would catch this, we'd be happy to merge it.

Comment (thefourtheye) [inline lib/outdated.js]: @othiym23 nvm. I was wrong. `a` and `b` are arrays in the form,

    [ '/home/thefourtheye/git/npm/test/tap/outdated',
      'underscore',
      '1.3.1',
      '1.3.1',
      '1.5.1',
      '1.3.1' ]

So, the code works fine only I guess.

Comment (othiym23) [inline lib/outdated.js]: I should have waited until I was less sleepy to take a look at this, so sorry for the erroneous feedback. I'm glad all is well!

**Pull request closed without merging**